pyo3-macros-backend/src/pyclass.rs

```diff
@@ -640,6 +770,164 @@ fn impl_enum(
     })
 }
 
+fn impl_complex_enum(
+    complex_enum: PyClassComplexEnum<'_>,
+    args: &PyClassArgs,
+    doc: PythonDoc,
+    methods_type: PyClassMethodsType,
+) -> Result<TokenStream> {
+    let krate = get_pyo3_crate(&args.options.krate);
+    let cls = complex_enum.ident;
+    let variants = complex_enum.variants;
+    let pytypeinfo = impl_pytypeinfo(cls, args, None);
+
+    let default_slots = vec![];
+
+    let pyclass_impls = PyClassImplsBuilder::new(
+        cls,
+        args,
+        methods_type,
+        complex_enum_default_methods(cls, variants.iter().map(|v| (v.ident, v.python_name(args)))),
+        default_slots,
+    )
+    .doc(doc)
+    .impl_all()?;
+
+    let variant_cls_newtypes = variants.iter().map(|variant| {
+        let variant_cls = gen_complex_enum_variant_class_ident(cls, variant.ident);
+        quote! {
+            #[doc(hidden)]
+            #[allow(non_camel_case_types)]
+            struct #variant_cls;
+        }
+    });
+
+    let variant_cls_pytypeinfos = variants.iter().map(|variant| {
+        let variant_cls = gen_complex_enum_variant_class_ident(cls, variant.ident);
+
+        // TODO(mkovaxx): propagate variant.options
+        let options: PyClassPyO3Options = parse_quote! {
```

Comment:
@davidhewitt This is how I'm trying to make the variant PyClasses frozen and subclass the enum PyClass, but it doesn't seem to be effective... I'm stuck, please help! 😅

Reply:
It looks like on line 818 below you call `impl_pytypeinfo` with the options, but you're building separate args with default options for `PyClassImplsBuilder` below. Maybe merge this loop with the one below and reuse the same options?

Reply:
Whoops, that was a silly mistake.
Thanks for pointing it out! :)